lib/gooddata/lcm/actions/collect_data_product.rb

```diff
@@ -0,0 +1,48 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class CollectDataProduct < BaseAction
+      DESCRIPTION = 'Collect DataProduct'
+
+      PARAMS = define_params(self) do
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'DataProduct to manage'
+        param :data_product, instance_of(Type::StringType), required: false
+      end
+
+      class << self
+        def call(params)
+          params = params.to_hash
+          domain = params[:GDC_GD_CLIENT].domain
+
+          if params.key?(:data_product)
+            data_product = domain.data_products(params[:data_product])
+            # kdyz neexistuje, tak vytvorime novy - jak z API poznam, ze neexistuje?
+          else
+            puts 'Please specify the data_product parameter when using the brick - trying to find a fallback'
+            data_products = domain.data_products(:all)
+            fail 'No data_product parameter specified - unable to fallback to a default' if data_products.length > 1 || data_products.empty?
```

Comment:
Is it possible to have `data_products.length == 0`? We should take this possibility into account.